src/components/Scene/CameraControls.component.tsx

```diff
@@ -54,60 +50,67 @@ export const CameraControls: FC<CameraControlsProps> = ({
 }) => {
   const cameraZoomTargetRef = useRef(cameraZoomTarget);
   const { camera, gl } = useThree();
+  const controlsRef = useRef<OrbitControls>(null);
+  const progressRef = useRef(Number.POSITIVE_INFINITY);
   const fallbackCameraTarget = cameraTarget || 1.475 + headScale / 10;
   const headScaleAdjustedMinDistance = controlsMinDistance + headScale / 10;
 
+  const updateCameraFocus = (delta: number, target?: Vector3) => {
+    if (target && progressRef.current <= 1) {
+      camera.position.setX(lerp(camera.position.x, target.x, progressRef.current));
+      camera.position.setZ(lerp(camera.position.z, target.z, progressRef.current));
+      progressRef.current += delta;
+    }
+  };
+
   useEffect(() => {
     if (
       cameraZoomTargetRef.current?.x !== cameraZoomTarget?.x ||
       cameraZoomTargetRef.current?.y !== cameraZoomTarget?.y ||
       cameraZoomTargetRef.current?.z !== cameraZoomTarget?.z
     ) {
       cameraZoomTargetRef.current = cameraZoomTarget;
-      progress = 0;
+      progressRef.current = 0;
     }
 
-    controls = new OrbitControls(camera, gl.domElement);
-    controls.enableRotate = false;
-    controls.enablePan = false;
-
-    controls.minDistance = headScaleAdjustedMinDistance;
-    controls.maxDistance = controlsMaxDistance;
-    controls.minPolarAngle = 1.4;
-    controls.maxPolarAngle = 1.4;
-
-    controls.target.set(0, fallbackCameraTarget, 0);
-    controls.update();
-
-    // TODO: Look for a better distance initialiser, without progress value check it conflicts with cameraZoomTarget which also can update camera position.z
-    if (cameraInitialDistance && progress === Number.POSITIVE_INFINITY) {
-      camera.position.z = cameraInitialDistance;
-      controls.update();
+    const controls = controlsRef.current;
+    if (controls) {
+      // TODO: Look for a better distance initialiser, without progress value check it conflicts with cameraZoomTarget which also can update camera position.z
+      if (cameraInitialDistance && progressRef.current === Number.POSITIVE_INFINITY) {
+        camera.position.z = cameraInitialDistance;
+        controls.update();
+      }
     }
 
     return () => {
-      controls.dispose();
+      controls?.dispose();
     };
```

Comment:
Just fyi it will be this return disposing of the controls after cameraZoomTarget has been updated